### ISSUE TITLE
[bruce/slam/feature_extraction-4] process has died [pid 28571, exit code -11, cmd

### ISSUE THREAD

**Issue by WLXM0922** [jake3991/sonar-SLAM#15]
Dear author, thank you very much for your work. But when I run roslaunch bruce_slam slam.launch and rosbag play sample_data.bag, I get an error: [bruce/slam/feature_extraction-4] process has died [pid 28571, exit code -11, cmd, Does anyone know why? I would appreciate it if you could answer it.

**Comment by jake3991** [comment]
Can you post the actual error message?

**Comment by WLXM0922** [comment]
> Can you post the actual error message?
of course, thank you very much for your reply. I get an error: [bruce/slam/feature_extraction-4] process has died [pid 62138, exit code -11, cmd /home/xlx/aa_ws/src/sonar-SLAM/bruce_slam/scripts/feature_extraction_node.py __name:=feature_extraction __log:=/home/xlx/.ros/log/eb3db3fe-f8c7-11ee-860f-bdcce353b5e6/bruce-slam-feature_extraction-4.log].
log file: /home/xlx/.ros/log/eb3db3fe-f8c7-11ee-860f-bdcce353b5e6/bruce-slam-feature_extraction-4*.log. The full contents of this log file are as follows: [rospy.client][INFO] 2024-04-12 20:26:43,424: init_node, name[/bruce/slam/feature_extraction], pid[62138]
[xmlrpc][INFO] 2024-04-12 20:26:43,424: XML-RPC server binding to 0.0.0.0:0
[xmlrpc][INFO] 2024-04-12 20:26:43,424: Started XML-RPC server [http://wl:43187/]
[rospy.init][INFO] 2024-04-12 20:26:43,424: ROS Slave URI: [http://wl:43187/]
[rospy.impl.masterslave][INFO] 2024-04-12 20:26:43,424: _ready: http://wl:43187/
[rospy.registration][INFO] 2024-04-12 20:26:43,425: Registering with master node http://localhost:11311
[xmlrpc][INFO] 2024-04-12 20:26:43,425: xml rpc node: starting XML-RPC server
[rospy.init][INFO] 2024-04-12 20:26:43,525: registered with master
[rospy.rosout][INFO] 2024-04-12 20:26:43,525: initializing /rosout core topic
[rospy.rosout][INFO] 2024-04-12 20:26:43,527: connected to core topic /rosout
[rospy.simtime][INFO] 2024-04-12 20:26:43,527: /use_sim_time is not set, will not subscribe to simulated time [/clock] topic
[rosout][INFO] 2024-04-12 20:26:43,534: Start online sonar feature extraction...
[rospy.internal][INFO] 2024-04-12 20:26:43,711: topic[/bruce/slam/feature_extraction/feature] adding connection to [/rviz], count 0
[rospy.internal][INFO] 2024-04-12 20:26:43,726: topic[/bruce/slam/feature_extraction/feature] adding connection to [/bruce/slam/slam], count 1
[rospy.internal][INFO] 2024-04-12 20:26:43,811: topic[/rosout] adding connection to [/rosout], count 0
[rospy.internal][INFO] 2024-04-12 20:26:43,812: topic[/bruce/slam/feature_extraction/feature_img] adding connection to [/rviz], count 0
[rospy.internal][INFO] 2024-04-12 20:27:04,158: topic[/sonar_oculus_node/ping] adding connection to [http://wl:37401/], count 0      Thank you very much for telling me what caused this.

**Comment by WLXM0922** [comment]
> Can you post the actual error message?
of course, thank you very much for your reply. I get an error: [bruce/slam/feature_extraction-4] process has died [pid 62138, exit code -11, cmd /home/xlx/aa_ws/src/sonar-SLAM/bruce_slam/scripts/feature_extraction_node.py __name:=feature_extraction __log:=/home/xlx/.ros/log/eb3db3fe-f8c7-11ee-860f-bdcce353b5e6/bruce-slam-feature_extraction-4.log].
log file: /home/xlx/.ros/log/eb3db3fe-f8c7-11ee-860f-bdcce353b5e6/bruce-slam-feature_extraction-4*.log. The full contents of this log file are as follows: [rospy.client][INFO] 2024-04-12 20:26:43,424: init_node, name[/bruce/slam/feature_extraction], pid[62138]
[xmlrpc][INFO] 2024-04-12 20:26:43,424: XML-RPC server binding to 0.0.0.0:0
[xmlrpc][INFO] 2024-04-12 20:26:43,424: Started XML-RPC server [http://wl:43187/]
[rospy.init][INFO] 2024-04-12 20:26:43,424: ROS Slave URI: [http://wl:43187/]
[rospy.impl.masterslave][INFO] 2024-04-12 20:26:43,424: _ready: http://wl:43187/
[rospy.registration][INFO] 2024-04-12 20:26:43,425: Registering with master node http://localhost:11311
[xmlrpc][INFO] 2024-04-12 20:26:43,425: xml rpc node: starting XML-RPC server
[rospy.init][INFO] 2024-04-12 20:26:43,525: registered with master
[rospy.rosout][INFO] 2024-04-12 20:26:43,525: initializing /rosout core topic
[rospy.rosout][INFO] 2024-04-12 20:26:43,527: connected to core topic /rosout
[rospy.simtime][INFO] 2024-04-12 20:26:43,527: /use_sim_time is not set, will not subscribe to simulated time [/clock] topic
[rosout][INFO] 2024-04-12 20:26:43,534: Start online sonar feature extraction...
[rospy.internal][INFO] 2024-04-12 20:26:43,711: topic[/bruce/slam/feature_extraction/feature] adding connection to [/rviz], count 0
[rospy.internal][INFO] 2024-04-12 20:26:43,726: topic[/bruce/slam/feature_extraction/feature] adding connection to [/bruce/slam/slam], count 1
[rospy.internal][INFO] 2024-04-12 20:26:43,811: topic[/rosout] adding connection to [/rosout], count 0
[rospy.internal][INFO] 2024-04-12 20:26:43,812: topic[/bruce/slam/feature_extraction/feature_img] adding connection to [/rviz], count 0
[rospy.internal][INFO] 2024-04-12 20:27:04,158: topic[/sonar_oculus_node/ping] adding connection to [http://wl:37401/], count 0      Thank you very much for telling me what caused this.

**Comment by jake3991** [comment]
I don't see any actual error messages there, is this the terminal you are running the bag in or the slam system?

**Comment by WLXM0922** [comment]
> I don't see any actual error messages there, is this the terminal you are running the bag in or the slam system?

When I run roslaunch bruce_slam.launch, rviz starts normally, and this terminal does not report an error. However, when I opened a new terminal to run osbag play sample_data.bag, the terminal running roslaunch bruce_slam.launch reported the above error.

**Comment by WLXM0922** [comment]
> I don't see any actual error messages there, is this the terminal you are running the bag in or the slam system?

I don't know if it's because of pcl.cpp and cfar.cpp in the directory of 'bruce_slam/src/bruce_slam/cpp/'. If these two files don't successfully create python interfaces through pybind11, will the above problems occur? I look forward to your reply very much and thank you very much.

**Comment by WLXM0922** [comment]
> I don't see any actual error messages there, is this the terminal you are running the bag in or the slam system?

Dear author, when I was debugging code in vscode, I ran slam.launch and the data set of a single robot. However, when feature extraction is carried out in the whole slam process, the feature_extraction.py file will crash in line 242nd code points = PCL. Downsample (points, self. resolution). What is the reason? I built it completely according to your project. Is it the pcl library version?

**Comment by yangyouze** [comment]
hello，have you solve this problem？ I meet the same problem，but i dont know how to solve it

**Comment by qimg412** [comment]
hello, I found that feature_extraction.py could not extract feature and the feature topic can not publish sucessfully, there was no data return
